### PR TITLE
refactor(@desktop/chat-community): `ChatContextMenuView` component updated

### PR DESCRIPTION
**Pull request by saledjenic**
- admin flag exposed to qml for chat/channel
- `CreateChannelPopup` updated
- `createPopup` function added to the `Global` singleton
- `PinnedMessagesPopup` component moved to `AppMain` and shared among components beneath

### CODE REVIEW

File: src/app/modules/main/chat_section/chat_content/messages/module.nim [hunk 1]
```diff
@@ -163,7 +164,15 @@ method getChatColor*(self: Module): string =
   return chatDto.color
 
 method amIChatAdmin*(self: Module): bool =
-  return false
+  if(not self.controller.belongsToCommunity()):
+    let chatDto = self.controller.getChatDetails()
+    for m in chatDto.members:
+      if (m.id == singletonInstance.userProfile.getPubKey()):
```
Comment (jrainville) [inline src/app/modules/main/chat_section/chat_content/messages/module.nim:170]: I guess this is for Chat Groups. Chat groups have a single admin. There should be a property for that. Being a member of a group does not make you an admin.

File: src/app/modules/main/chat_section/chat_content/module.nim [hunk 2]
```diff
@@ -216,4 +229,18 @@ method onReactionRemoved*(self: Module, messageId: string, emojiId: int, reactio
...
+    error "(pinned) wrong emoji id found on reaction remove response", emojiId
+
+method getCurrentFleet*(self: Module): string =
+  return self.controller.getCurrentFleet()
+
+method amIChatAdmin*(self: Module): bool =
+  if(not self.controller.belongsToCommunity()):
+    let chatDto = self.controller.getChatDetails()
+    for m in chatDto.members:
+      if (m.id == singletonInstance.userProfile.getPubKey()):
```
Comment (jrainville) [inline src/app/modules/main/chat_section/chat_content/module.nim:241]: same thing here

File: src/app/modules/main/chat_section/module.nim [hunk 3]
```diff
@@ -41,36 +44,49 @@ proc newModule*(
...
   self.view.delete
   self.viewVariant.delete
   self.controller.delete
 
 method isCommunity*(self: Module): bool =
   return self.controller.isCommunity()
 
+proc amIMarkedAsAdminUser(self: Module, members: seq[ChatMember]): bool = 
+  for m in members:
+    if (m.id == singletonInstance.userProfile.getPubKey()):
```
Comment (jrainville) [inline src/app/modules/main/chat_section/module.nim:66]: Probably same

File: src/app/modules/main/module.nim [hunk 4]
```diff
@@ -189,7 +192,7 @@ method load*[T](
   let (unviewedCount, mentionsCount) = self.controller.getNumOfNotificaitonsForChat()
   let hasNotification = unviewedCount > 0 or mentionsCount > 0
   let notificationsCount = mentionsCount
-  let chatSectionItem = initItem(conf.CHAT_SECTION_ID, SectionType.Chat, conf.CHAT_SECTION_NAME, "", "",
+  let chatSectionItem = initItem(conf.CHAT_SECTION_ID, SectionType.Chat, conf.CHAT_SECTION_NAME, false, "", "",
```
Comment (jrainville) [inline src/app/modules/main/module.nim:195]: Suggestion: for all those empty or boolean values, it's nice to have the property name, like:
```nim
initItem(conf.CHAT_SECTION_ID, SectionType.Chat, conf.CHAT_SECTION_NAME, isAdmin = false, name = "", whatever = "",
```
disclaimer: I don't know the names of the properties, but it's just easier to understand for later

**Review by jrainville — COMMENTED** [review]
Nice job.

I think the admin thing is wrong though. It seems like you check if a user is a member of group, it counts as an admin, but there should be only one admin.
Let me know if I misunderstood.

**Comment by saledjenic** [comment]
@jrainville you're totally right, I added `admin` to the `dto`, but forgot to add it into condition. :) Thanks.

And I agree about adding names for `initItem`, so we don't have to switch to `section_item.nim` in order to see what param is what all the time. 

All updated.

File: src/app/modules/main/chat_section/chat_content/messages/module.nim [hunk 5]
```diff
@@ -163,7 +164,15 @@ method getChatColor*(self: Module): string =
   return chatDto.color
 
 method amIChatAdmin*(self: Module): bool =
-  return false
+  if(not self.controller.belongsToCommunity()):
+    let chatDto = self.controller.getChatDetails()
+    for m in chatDto.members:
+      if (m.id == singletonInstance.userProfile.getPubKey() and m.admin):
```
Comment (jrainville) [inline src/app/modules/main/chat_section/chat_content/messages/module.nim:170]: I think the admin is always part of the members, so the check might be redundant here, but it's not a big deal

**Review by jrainville — APPROVED** [review]
Good job. LGTM

File: src/app/modules/main/chat_section/chat_content/messages/module.nim (hunk 5 above)
Comment (saledjenic) [reply src/app/modules/main/chat_section/chat_content/messages/module.nim:170]: @jrainville not sure what is redundant here?
So `chatDto.members` contains list of users with properties `id` and `admin`, myself as logged in user may be in a members list but `admin` flag for me may be either `true` or `false`, that means I am an admin of this chat if `m.id` is set to my public key and `m.admin` is set to `true`.

Comment (jrainville) [reply src/app/modules/main/chat_section/chat_content/messages/module.nim:170]: Oops, my bad, I misread. I had not realized that `m` came from the loop. Ignore my comment

**Review by borismelnik — APPROVED** [review]

**Pull request merged**